### PR TITLE
[TechDocs] Set entity triplets to lowercase if config is set on docs tab at entity page

### PR DESCRIPTION
**Pull request by mertcb**
Co-authored-by: Mehmet Mallı <mehmet.malli@trendyol.com>
Signed-off-by: mertcbilgic <mert.bilgic@trendyol.com>

## Hey, I just made a Pull Request!

This PR fixes the issue at #8381 .  It sets entity triplets to lowercase if the config is set on docs tab at entity page.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs/src/EntityPageDocs.tsx [hunk 1]
```diff
@@ -17,13 +17,21 @@
...
 import { Reader } from './reader';
+import { configApiRef, useApi } from '@backstage/core-plugin-api';
 
 export const EntityPageDocs = ({ entity }: { entity: Entity }) => {
+  // Lower-case entity triplets by default, but allow override.
+  const toLowerMaybe = useApi(configApiRef).getOptionalBoolean(
+    'techdocs.legacyUseCaseSensitiveTripletPaths',
+  )
+    ? (str: string) => str
+    : (str: string) => str.toLocaleLowerCase('en-US');
```
Comment (emmaindal) [inline plugins/techdocs/src/EntityPageDocs.tsx:28]: Now when we have a third duplicate of this implemented, can we add this as a helper function at `/src` and use the same across the EntityPageDocs, [DocsCardGrid](https://github.com/backstage/backstage/blob/7439368ce532e98bea8994b2be1899c9af9626ca/plugins/techdocs/src/home/components/DocsCardGrid.tsx) and [DocsTable](https://github.com/backstage/backstage/blob/7439368ce532e98bea8994b2be1899c9af9626ca/plugins/techdocs/src/home/components/DocsTable.tsx)

File: plugins/techdocs/src/EntityPageDocs.tsx [hunk 2]
```diff
@@ -17,13 +17,21 @@
...
+
   return (
     <Reader
       withSearch={false}
       entityRef={{
-        kind: entity.kind,
+        kind: toLowerMaybe(entity.kind),
         namespace: entity.metadata.namespace ?? 'default',
         name: entity.metadata.name,
```
Comment (emmaindal) [inline plugins/techdocs/src/EntityPageDocs.tsx:36]: I believe we want to run namespace and name through it as well, similar to how its used here https://github.com/backstage/backstage/blob/7439368ce532e98bea8994b2be1899c9af9626ca/plugins/techdocs/src/home/components/DocsTable.tsx#L70-L72

Comment (emmaindal) [reply plugins/techdocs/src/EntityPageDocs.tsx:36]: if you update this you can also take the opportunity to update the name of the changeset to be prefixed with techdocs-*, that way we can focus on the techdocs-core team to review it!

File: plugins/techdocs/src/EntityPageDocs.tsx (hunk 1 above)
Comment (mertcb) [reply plugins/techdocs/src/EntityPageDocs.tsx]: done that emma :)

**Review by emmaindal — APPROVED** [review]
Amazing, thank you very much! @mertcb

**Pull request merged**